Doc/library/urllib.request.rst

```diff
@@ -199,7 +199,11 @@ The following classes are provided:
    bytes objects, while ``Transfer-Encoding: chunked`` as specified in
    :rfc:`7230`, Section 3.3.1 will be used to send files and other iterables.
 
-   For an HTTP POST request method, *data* should be a buffer in the
+   For an HTTP POST request method, *data* must be either
+   :term:`bytes-like object <bytes-like object>` or an iterable of it,
+   including a dict with key of this type (this is not supported
+   for HTTPS), and in that case Content-Length value must be specified
```

Comment:
Nan’s patch listed supported data types in general, for all HTTP request methods. These types should work for other HTTP methods, not just POST.

I expect the problem with HTTPS is fixed now; see <https://bugs.python.org/issue27340>.

Content-Length is no longer required; read the previous paragraph.